src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Internal adapter for adding custom functions. 
+    /// </summary>
+    internal class CustomTexlFunction : TexlFunction
+    {
+        public Func<IServiceProvider, FormulaValue[], CancellationToken, Task<FormulaValue>> _impl;
+
+        internal BigInteger LamdaParamMask;
+
+        private readonly IEnumerable<CustomFunctionSignatureHelper> _argumentSignatures;
```

Comment:
>_argumentSignatures; [](http://example.com/codeflow?start=68&length=20)

This holds custom signatures,
each entry would map to a single overload.

If function doesn't have any overload, this will have single entry.